src/lib/storage/table.cpp

```diff
@@ -80,4 +84,16 @@ Chunk& Table::get_chunk(ChunkID chunk_id) { return _table_chunks.at(chunk_id); }
 
 const Chunk& Table::get_chunk(ChunkID chunk_id) const { return _table_chunks.at(chunk_id); }
 
+void Table::compress_chunk(ChunkID chunk_id) {
+  Chunk compressed_chunk;
+  Chunk& old_chunk = _table_chunks.at(chunk_id);
+
+  for (int i = 0; i < old_chunk.col_count(); i++) {
+    const auto col = make_shared_by_column_type<BaseColumn, DictionaryColumn>(column_type(ColumnID(i)),
+                                                                              old_chunk.get_column(ColumnID(i)));
+    compressed_chunk.add_column(col);
+  }
+  _table_chunks[chunk_id] = std::move(compressed_chunk);
```

Comment:
consider using `_table_chunks.at(chunk_id)` - mostly for continuity and coherence, as the previous checks already make sure this shouldn't run out of bounds.